app/admin/page.tsx

```diff
@@ -18,9 +18,18 @@ export default function Admin() {
   const [rules, setRules] = useState<RuleInfo[]>([]);
 
   const getOrRefreshRuleList = async () => {
-    const data = await getAllRuleData();
-    setInitialRules(data);
-    setRules(JSON.parse(JSON.stringify(data))); // JSON.parse(JSON.stringify(data)) is a hacky way to deep copy the data - needed for comparison later
+    // Get rules that are already defined in the DB
+    const existingRules = await getAllRuleData();
+    setInitialRules(existingRules);
+    // Get rules that exist in the rules repository, but aren't yet defined in the DB
+    const existingRuleDocuments = await getAllRuleDocuments();
+    const undefinedRules = existingRuleDocuments
+      .filter((ruleJSON: string) => {
+        return !existingRules.find((rule: RuleInfo) => rule.goRulesJSONFilename === ruleJSON);
+      })
+      .map((ruleJSON: string) => ({ goRulesJSONFilename: ruleJSON }));
+    const ruleData = [...existingRules, ...undefinedRules];
```

Comment:
This allows us to map rules from the rules repository to the db in the admin page. I imagine we will probably remove this page entirely at some point, but this works for now until we do.